class.coupon.php

```diff
@@ -124,7 +124,7 @@ static private function cleanString($string, $options = []) {
         $toUpper = (isset($options['uppercase']) ? filter_var($options['uppercase'], FILTER_VALIDATE_BOOLEAN) : false);
         $toLower = (isset($options['lowercase']) ? filter_var($options['lowercase'], FILTER_VALIDATE_BOOLEAN) : false);
 
-        $striped = preg_replace('/[^a-zA-Z0-9]/', '', trim($string));
+        $striped = preg_replace('/[^a-zA-Z0-9-]/', '', trim($string));
```

Comment:
if ($toLower && $toUpper) {
            throw new Exception('You cannot set both options (uppercase|lowercase) to "true"!');
        } else if ($toLower) {
            return strtolower($striped);
        } else if ($toUpper) {
            return strtoupper($striped);
        } else {
            return $striped;
        }
    }
}